aws/resource_aws_launch_template.go

```diff
@@ -984,9 +984,14 @@ func readNetworkInterfacesFromConfig(ni map[string]interface{}) *ec2.LaunchTempl
 	var ipv4Addresses []*ec2.PrivateIpAddressSpecification
 	var ipv6Addresses []*ec2.InstanceIpv6AddressRequest
 	var privateIpAddress string
-	networkInterface := &ec2.LaunchTemplateInstanceNetworkInterfaceSpecificationRequest{
-		AssociatePublicIpAddress: aws.Bool(ni["associate_public_ip_address"].(bool)),
-		DeleteOnTermination:      aws.Bool(ni["delete_on_termination"].(bool)),
+	networkInterface := &ec2.LaunchTemplateInstanceNetworkInterfaceSpecificationRequest{}
+
+	if v, ok := ni["associate_public_ip_address"]; ok && v.(bool) {
```

Comment:
I believe setting `false` should be a valid value (at least it is in the EC2 console from the options of don't set / disable / enable) -- I think we need to cover both cases of ignoring this attribute when `network_interface_id` is set, as well as allowing `false`.